p2p/src/main/java/bisq/network/p2p/storage/payload/ProtectedStorageEntry.java

```diff
@@ -41,25 +43,29 @@
     private int sequenceNumber;
     private byte[] signature;
     private long creationTimeStamp;
+    transient private final Clock clock;
```

Comment:
There is an alternative here that doesn't involve storing the clock inside the data structure. The Public API methods that need a clock (ctor/refresh/backdate/isExpired) can take one as a parameter for the work they need individually. This will involve plumbing a Clock into OfferBookService, but that is easy enough.

The protobuf deserialization path still needs access to a clock, so it will still need the resolver instance where it can be used in maybeAdjustCreationTimeStamp().

If someone has a strong opinion on this or maybe @christophsturm can comment on what he envisioned with the clock injections throughout the code. I'd rather stick to a common pattern to make this work consistent in the future.

Reply:
I did not have time to look closer but adding Clock to ProtectedStorageEntry does not feel right to me. Using injecting for service classes is fine and adding clock there, but I think it should not be part of the value objects. 
Those methods on ProtectedStorageEntry using time could be moved to the service class and the ProtectedStorageEntry is passed as an object (just a rough idea, had not looked closer to the PR...). Those heavy value objects doing too much and having external dependencies (time) have been a bad idea anyway and maybe thats a good opportunity to refactor that.

Reply:
After sleeping on it and coming back to it today, I don't like the idea of storing the clock in the object. Not necessarily because it is value-heavy, but because it is more testable and cleaner to pass in a new instance if needed instead of relying on the one that was generated at instantiation time

I've updated the stack with the patch that removes it in favor of passing it into the functions that need it.

After reviewing and refactoring this code over the last few weeks I am actually of the opinion that this needs more OO not less to stay maintainable and testable. Having one large integration test for everything isn't a great path forward. I've got a stack almost ready that leverages the inheritance that already exists to make the code a lot easier test and deduplicates a lot of the redundant helper functions.

I'll send it out tomorrow and whoever is interested can take a look and see if the pros outweigh the cons.

Thanks again for your feedback.

Reply:
This has been fixed. ProtectedStorageEntry takes in a Clock in all usages instead of storing it.